climada/util/api_client.py

```diff
@@ -486,3 +493,145 @@ def purge_cache(self, local_path):
+          ----------
+          country : list
+              List of country for which to create the LitPop object. If None is given, a global LitPop instance
+              is created
+          data_dir : str
+              directory where the files should be downoladed. Default: SYSTEM_DIR
+          """
+        if not country:
+            datasets = [self.get_datasets(data_type='litpop', properties={'exponents': '(1,1)', 'fin_mode': 'pc'})]
+        if country:
+            country = [country]
+            datasets = [self.get_dataset(data_type='litpop', properties={'exponents': '(1,1)', 'fin_mode': 'pc',
+                                                                             'country_name': c}) for c in country]
+        exposures_list = []
+        for dataset in datasets:
+            if os.path.isfile(os.path.join(data_dir, dataset.files[0].file_name)):
+                LOGGER.info('The file already exists and it was not downloaded again.')
+            self.download_file(data_dir, dataset.files[0])
+            exposures = Exposures()
+            exposures.read_hdf5(os.path.join(data_dir, dataset.files[0].file_name))
+            exposures_list.append(exposures)
+        exposures_concat = Exposures()
+        exposures_concat = exposures_concat.concat(exposures_list)
+        return exposures_concat
+
+    def _get_data(self, type):
+        datasets = self.get_datasets(data_type=type)
+        properties_keys = np.unique([dataset.properties.keys() for dataset in datasets])
+        # find common properties between "groups" of datasets:
+        properties_keys = set(properties_keys[0]).intersection(*properties_keys)
+        user_properties_input = {}
+        # get user input to differentiate between these groups
+        user_properties_input = self._select_properties(datasets, properties_keys, user_properties_input)
+        datasets = self.get_datasets(data_type=type, properties=user_properties_input)
+        # find remaining properties to be chosen (usually countries:)
+        properties_keys2 = set(np.unique([dataset.properties.keys() for dataset in datasets])[0]) - properties_keys
+        user_properties_input.update(self._select_properties(datasets, properties_keys2, user_properties_input))
+        try: # make a list of properties and get a list of datasets in case several countries are given
+            user_properties_input['country_name']
+            datasets_properties = []
+            for country in user_properties_input['country_name']:
+                properties = user_properties_input.copy()
+                properties['country_name'] = country
+                datasets_properties.append(properties)
+        except:
+            datasets_properties = [user_properties_input]
+        datasets = [self.get_dataset(data_type=type, properties=properties) for properties in datasets_properties]
+        return datasets
+
+    def _select_properties(self, datasets, properties_keys, user_properties_input):
```

Comment:
I still have to make sure that only possible combinations of properties are suggested here